Make sure the StatusCode is taken into account in DoRaw()

asalkeld opened the pull request:
**What this PR does / why we need it**:
Currently if there is an error (not found) the error printed out
is to do with the inablity to convert an empty body into the expected json.

This patch will fill in the err correctly.

example of before (with NotFound error):
$ kubectl top node
failed to unmarshall heapster response: json: cannot unmarshal object into Go value of type []v1alpha1.NodeMetrics

Now:
$ kubectl top node
the server could not find the requested resource (get services http:heapster:)

**Which issue this PR fixes** 
related to bug #30818

**Special notes for your reviewer**:
None

**Release note**:

``` release-note
NONE
```

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.kubernetes.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/30969)

<!-- Reviewable:end -->

Commit:
Make sure the StatusCode is taken into account in DoRaw()

Currently if there is an error (not found) the error printed out
is to do with the inablity to convert an empty body into the expected json.

This patch will fill in the err correctly.

example of before (with NotFound error):
$ kubectl top node
failed to unmarshall heapster response: json: cannot unmarshal object into Go value of type []v1alpha1.NodeMetrics

Now:
$ kubectl top node
the server could not find the requested resource (get services http:heapster:)

krousey commented:
Reviewed 1 of 1 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[test/e2e/networking.go, line 62 \[r2\]](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/30969#-KPYj-bt1JFkf_td8tBg:-KPYj-buT3K_dXhdrqDZ:bq85wkp) ([raw file](https://github.com/kubernetes/kubernetes/blob/1e09eb79492a1e0fc2df534fe6d80f6696fd5408/test/e2e/networking.go#L62)):*
> ```Go
> 			path string
> 		}{
> 			{path: "/validate"},
> ```

Why is this being removed?

---


*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/30969)*
<!-- Sent from Reviewable.io -->

Commit:
Fix static URL conformance test

1. /validate service does not exist, so remove the test for it and add some that actually do exist
2. The namespace does not exist so this will always return NotFound

Note: DoRaw() ignores the StatusCode.
This is in preparation for the next commit

asalkeld commented:
Review status: 0 of 2 files reviewed at latest revision, 1 unresolved discussion.

---

*[test/e2e/networking.go, line 62 \[r2\]](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/30969#-KPYj-bt1JFkf_td8tBg:-KPk31yYPMXa2sjQ1WMh:b-9ex2gx) ([raw file](https://github.com/kubernetes/kubernetes/blob/1e09eb79492a1e0fc2df534fe6d80f6696fd5408/test/e2e/networking.go#L62)):*
<details><summary><i>Previously, krousey wrote…</i></summary>
> Why is this being removed?
</details>
It does not exist. All requests to /validate fail with NotFound.

---


*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/30969)*
<!-- Sent from Reviewable.io -->

krousey commented:
Review status: 0 of 2 files reviewed at latest revision, 1 unresolved discussion.

---

*[test/e2e/networking.go, line 71 \[r4\]](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/30969#-KPnKXlqg18U8w3pAnbL:-KPnKXlrQ4xpkHt7qJUZ:b-qbjytv) ([raw file](https://github.com/kubernetes/kubernetes/blob/d5a13ad420b783b9ce60bcbd655b38694b3a2716/test/e2e/networking.go#L71)):*
> ```Go
> 			// TODO: test proxy links here
> 		}
> 		//				Namespace(f.Namespace.Name).
> ```

Delete this comment

---


*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/30969)*
<!-- Sent from Reviewable.io -->

asalkeld commented:
Review status: 0 of 2 files reviewed at latest revision, 1 unresolved discussion.

---

*[test/e2e/networking.go, line 71 \[r4\]](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/30969#-KPnKXlqg18U8w3pAnbL:-KPp3XKEacNh514Jb2lw:b-p1sahv) ([raw file](https://github.com/kubernetes/kubernetes/blob/d5a13ad420b783b9ce60bcbd655b38694b3a2716/test/e2e/networking.go#L71)):*
<details><summary><i>Previously, krousey wrote…</i></summary>
> Delete this comment
</details>
Will do, thanks for spotting that.

---


*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/30969)*
<!-- Sent from Reviewable.io -->

krousey commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.kubernetes.io/lgtm.png" height="20" width="61"/>

---

Reviewed 1 of 2 files at r5, 1 of 1 files at r6.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/30969#-:-KPs8jn0Q5vSkLDMhJ9F:bnfp4nl)*
<!-- Sent from Reviewable.io -->
